plan/new.go

```diff
@@ -52,6 +57,20 @@ func buildTestCases(plan *Plan) []TestCase {
 				Plan:      plan,
 				Client:    client,
 				Arguments: testArgs,
+				Skip:      false,
+			}
+
+			for _, filter := range behavior.Filters {
+				// Each filter should be a complete match.
+				if filter.Matches(testArgs) {
+					t.Skip = true
+					var formattedMatches []string
+					for _, match := range filter.Matchers {
+						formattedMatches = append(formattedMatches, fmt.Sprintf("%s:%s", match.Name, match.Value))
+					}
```

Comment:
This could be greatly simplified by a `String()` function on `AxisMatcher` and another on `Filter`.